Foundation/URLCache.swift

```diff
@@ -173,13 +217,102 @@ open class URLCache : NSObject {
     */
     open class var shared: URLCache {
         get {
-            NSUnimplemented()
+            return sharedLock.performLocked {
+                return _shared
+            }
         }
         set {
-            NSUnimplemented()
+            sharedLock.performLocked {
+                _shared = newValue
+            }
         }
     }
-
+    
+    private let cacheDirectory: URL?
+    
+    private struct CacheEntry: Hashable {
+        var identifier: String
```

Comment:
@drodriguez Generally, I like keeping struct entries `var` unless there's a strict need; value type semantics prevent unintended mutation at usage sites anyway, and more complex mutation scenarios are aided by the ability to copy a value and edit the fields.